# SEE THE README.MD FOR ARGUMENT DESCRIPTIONS

  
- name: "/topological_navigation/feedback"
#  signal_when: 
#    condition: "not published"
#    timeout: 0.0
#    safety_critical: False
#    process_indices: []
#    repeat_exec: False
  signal_lambdas:
  - expression: "lambda msg: msg.feedback.route == 'WayPoint1'"
    timeout: 0.0
    safety_critical: True
    when_published: False
#    process_indices: [0,1,2,3,4]
    repeat_exec: False
  execute:
  - lock_acquire:
  - log:
      message: "Navigating away from {}"
      level: warn
      msg_args:
      - "msg.feedback.route"
  - action:
      verbose: True
      namespace: "/topological_navigation"
      package: "topological_navigation"
      action_spec: "GotoNodeAction"
      goal_args:
      -  "goal.target = 'WayPoint45'"
  - reconf:
      verbose: True
      params:
      - ns: row_detector
        name: position_error_threshold
        value: -1
  - log:
      message: "Row detection disabled"
      level: warn
  - shell:
      verbose: True
      cmd_args:
      - "cowsay"
      - "moo"
  - lock_release:
  timeout: 0.0
  default_notifications: True
  include: True                  


- name: "/topological_navigation/feedback"
#  signal_when:
#    condition: "not published"
#    timeout: 0.0
#    safety_critical: False
#    process_indices: []
#    repeat_exec: False
  signal_lambdas:
  - expression: "lambda msg: msg.feedback.route == 'WayPoint45'"
    timeout: 0.0
    safety_critical: False
    when_published: False
#    process_indices: [0,1,2,3,4,5,6,7]
    repeat_exec: False
  execute:
  - reconf:
      verbose: True
      params:
      - ns: row_detector
        name: position_error_threshold
        value: 0.3
  - log:
      message: "Row detection enabled"
      level: info
  - call:
      verbose: True
      service_name: "/sentor/set_safety_tag"
      service_args:
      -  "req.data = True"
  - log:
      message: "Teleporting the robot"
      level: info
  - call:
      verbose: True
      service_name: "/gazebo/set_model_state"
      service_args:
      -  "req.model_state.model_name = 'thorvald_ii'"
      -  "req.model_state.pose.orientation.w = 1.0"
  - sleep:
      verbose: True
      duration: 3.0
  - log:
      message: "Relocalising the robot"
      level: info
  - publish:
      verbose: True
      topic_name: "/initialpose"
      topic_latched: False
      topic_args:
      -  "msg.header.frame_id = '/map'"
      -  "msg.pose.pose.orientation.w = 1.0"
  timeout: 0.0
  default_notifications: True
  include: True   


- name : '/row_detector/path_error'
  signal_lambdas :
  - expression: "lambda msg : math.isnan(msg.y)"
  include: False 

